### ISSUE TITLE
What is the exact semantics of `gphase()` ?

### ISSUE THREAD

**Issue by nakrakiiya** [openqasm/openqasm#383]
What is the exact semantics of `gphase()` ? The paper and spec only said `adds a global phase of e^(iγ) to the scope containing the instruction.`

For example, the following code

```
gate foo q0,q1,q2,q3 {
  gphase(1);
  U(0,1,1) q0;
  U(0,1,2) q0;

  gphase(2);
  U(0,2,1) q1;
 
  gphase(3);
  cx q2, q3;
}
```

What is the matrix of gate `foo` for q0, q1, and q2~q3?

**Comment by stevenheidel** [comment]
I'm not sure if we ever defined what happens when multiple global phases are added. There are three choices for your example:
1) gphase(3), ie the last gphase takes precedence
2) gphase(6), ie global phases are multiplied together (exp(1i) * exp(2i) * exp(3i))
3) it's an error

**Comment by nakrakiiya** [comment]
> I'm not sure if we ever defined what happens when multiple global phases are added. There are three choices for your example:
> 
>     1. gphase(3), ie the last gphase takes precedence
> 
>     2. gphase(6), ie global phases are multiplied together (exp(1i) * exp(2i) * exp(3i))
> 
>     3. it's an error

I've studied the implementation in Amazon Braket. `gphase(x)` in Braket works as an n-qubits global phase gate applies in all qubits in current scope. There's no limit on where and how many times it appears.

The n-qubits global phase gate is:

$$ 
\text{gphase}_n(x) = e^{i x} I _ {2^n}
$$



I think the semantics of `gphase()` should be more specific in the spec.

**Comment by stevenheidel** [comment]
The TSC discussed today and agreed that gphases should accumulate. @nakrakiiya would you like to open a PR and add a sentence to the spec outlining this detail?

**Comment by hodgestar** [comment]
This appears to have been addressed in #393.

**Comment by hodgestar** [comment]
Closing as addressed by #393. Please feel free to comment if anything is still unclear.
